src/main/scala/org/broadinstitute/dsde/firecloud/service/ExportEntitiesByTypeActor.scala

```diff
@@ -1,79 +1,224 @@
+
+  lazy val logger: Logger = LoggerFactory.getLogger(getClass)
+
+  /**
+    * Two basic code paths
+    *
+    * For Collection types, write the content to temp files, zip and return.
+    *
+    * For Singular types, pipe the content from `Source` -> `Flow` -> `Sink`
+    *   Source generates the entity queries
+    *   Flow executes the queries and sends formatted content to chunked response handler
+    *   Sink finishes the execution pipeline
+    *
+    * Handle exceptions directly by completing the request.
+    */
+  def streamEntities(ctx: RequestContext, workspaceNamespace: String, workspaceName: String, entityType: String, attributeNames: Option[IndexedSeq[String]]): Future[Any] = {
+
+    getEntityTypeMetadata(workspaceNamespace, workspaceName, entityType) flatMap { metadata =>
+      val entityQueries = getEntityQueries(metadata, entityType)
+      if (TSVFormatter.isCollectionType(entityType)) {
+        streamCollectionType(ctx, workspaceNamespace, workspaceName, entityType, entityQueries, metadata, attributeNames)
+      } else {
+        val headers = TSVFormatter.makeEntityHeaders(entityType, metadata.attributeNames, attributeNames)
+        streamSingularType(ctx, workspaceNamespace, workspaceName, entityType, entityQueries, metadata, headers, attributeNames)
+      }
+    }
+  }.recoverWith {
+    case f: FireCloudExceptionWithErrorReport =>
+      logger.error(s"FireCloudExceptionWithErrorReport: Error generating entity download for $workspaceNamespace:$workspaceName:$entityType")
+      Future(ctx.complete(HttpResponse(
+        status = f.errorReport.statusCode.getOrElse(StatusCodes.InternalServerError),
+        entity = HttpEntity(ContentTypes.`application/json`, f.errorReport.toJson.compactPrint))))
+    case t: Throwable =>
+      logger.error(s"Throwable: Error generating entity download for $workspaceNamespace:$workspaceName:$entityType")
+      val errorReport = ErrorReport(StatusCodes.InternalServerError, s"Error generating entity download for $workspaceNamespace:$workspaceName:$entityType " + t.getMessage)
+      Future(ctx.complete(HttpResponse(
+        status = StatusCodes.InternalServerError,
+        entity = HttpEntity(ContentTypes.`application/json`, errorReport.toJson.compactPrint))))
+  }
+
+
+  /*
+   * Helper Methods
+   */
+
+
+  /**
+    * General Approach
+    * 1. Define a `Source` of entity queries
+    * 2. Run the source events through a `Flow`.
+    * 3. Flow sends events (batch of entities) to a streaming output actor
+    * 4. Return a Done to the calling route when complete.
+    */
+  private def streamSingularType(ctx: RequestContext, workspaceNamespace: String, workspaceName: String, entityType: String, entityQueries: Seq[EntityQuery], metadata: EntityTypeMetadata, headers: IndexedSeq[String], attributeNames: Option[IndexedSeq[String]]): Future[Done] = {
+    // Akka Streams Support
+    implicit val system: ActorSystem = ActorSystem("Streaming-Entity-Exporter")
```

Comment:
Was able to get rid of `ActorSystem` by pulling helpers into actor class. Looking into moving `ActorMaterializer` to the app level. That's only used by `akka.stream` so we have never needed that before.